inc/pendingreason_item.class.php

```diff
@@ -0,0 +1,520 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+class PendingReason_Item extends CommonDBRelation
+{
+   public static $itemtype_1 = 'PendingReason';
+   public static $items_id_1 = 'pendingreasons_id';
+   public static $take_entity_1 = false;
+
+   public static $itemtype_2 = 'itemtype';
+   public static $items_id_2 = 'items_id';
+   public static $take_entity_2 = true;
+
+   public static function getTypeName($nb = 0) {
+      return _n('Item', 'Items', $nb);
+   }
+
+   public static function getForItem(CommonDBTM $item) {
+      $em = new self();
+      $find = $em->find([
+         'itemtype' => $item::getType(),
+         'items_id' => $item->getID(),
+      ]);
+
+      if (!count($find)) {
+         return false;
+      }
+
+      $row_found = array_pop($find);
+      return self::getById($row_found['id']);
+   }
+
+   public static function createForItem(CommonDBTM $item, array $fields): bool {
+      $em = new self();
+      $find = $em->find([
+         'itemtype' => $item::getType(),
+         'items_id' => $item->getID(),
+      ]);
+
+      if (count($find)) {
+         // Clean existing entry
+         Toolbox::logError("FOUND one, DELETING", $item);
+         $to_delete = array_pop($find);
+         $fields['id'] = $to_delete['id'];
+         $em->delete(['id' => $fields['id']]);
+         unset($fields['id']);
+         $em = new self();
+      }
+
+      $fields['itemtype'] = $item::getType();
+      $fields['items_id'] = $item->getID();
+      $fields['last_bump_date'] = date("Y-m-d H:i:s");
```

Comment:
Nice, didn't knew this existed, done in 1bdf36217d.

Not relevant to this PR but maybe it would be nice to have a static method to access it ?
This way we have auto complete instead of having to "know" that this specific key exist in the session.

Also I find around 50+ occurrences of `date("Y-m-d H:i:s")` in the code, maybe they should be cleaned too in another PR.